tests/StorageApiTestCase.php

```diff
@@ -780,17 +783,17 @@ protected function listEvents(Client $client, $eventName, $expectedObjectId = nu
      */
     protected function retry($apiCall, $retries, $eventName)
     {
-        $events = [];
-        while ($retries > 0) {
+        $retryPolicy = new CallableRetryPolicy(function () use ($apiCall, $eventName) {
             $events = $apiCall();
             if (empty($events) || $events[0]['event'] !== $eventName) {
-                $retries--;
-                usleep(250 * 1000);
-            } else {
-                break;
+                return true;
             }
-        }
-        return $events;
+            return false;
+        }, $retries);
+        $proxy = new RetryProxy($retryPolicy, new FixedBackOffPolicy(250));
+        return $proxy->call(function () use ($apiCall) {
+            return $apiCall();
```

Comment:
Přišlo by mi jednodušší mít tady $events = apiCall(); if (!$events) throw

Tu exception by to chytlo a retrynulo pomocí klidně SimpleRetry policy. 

Přijde mi divný volat ten api call v RetryPolicy - tu retryovanou činnost bys měl dělat jen v tom call... 

Takhle, jak to je teď, by mimochodem mohl být parametr jen ten $apiCall bez ničeho dalšího.